File: embedded-continuous-query-update-support/src/main/scala/org/littlewings/infinispan/continuousquery/IndexedBook.scala
package org.littlewings.infinispan.continuousquery

import org.apache.lucene.analysis.standard.StandardAnalyzer
import org.hibernate.search.annotations._

import scala.beans.BeanProperty

object IndexedBook {
  def apply(isbn: String, title: String, price: Int): IndexedBook = {
    val book = new IndexedBook
    book.isbn = isbn
    book.title = title
    book.price = price
    book
  }
}

@Indexed
@Analyzer(impl = classOf[StandardAnalyzer])
@SerialVersionUID(1L)
class IndexedBook extends Serializable {
  @Field(analyze = Analyze.NO)
  @BeanProperty
  var isbn: String = _

  @Field
  @BeanProperty
  var title: String = _

  @Field
  @BeanProperty
  var price: Int = _

  def newBookChangePrice(newPrice: Int): IndexedBook = {
    val book = new IndexedBook
    book.isbn = isbn
    book.title = title
    book.price = newPrice
    book
  }
}

File: remote-support-proto3-record/src/main/java/org/littlewings/infinispan/remote/proto3record/IndexedBook.java
package org.littlewings.infinispan.remote.proto3record;

import org.infinispan.api.annotations.indexing.Basic;
import org.infinispan.api.annotations.indexing.Indexed;
import org.infinispan.api.annotations.indexing.Keyword;
import org.infinispan.api.annotations.indexing.Text;
import org.infinispan.protostream.annotations.Proto;

@Proto
@Indexed
public record IndexedBook(
        @Keyword(sortable = true)
        String isbn,
        @Text(analyzer = "standard")
        String title,
        @Basic(sortable = true)
        int price
) {
}
